From fcadf33e2352c8e56d6c4fc19ea1342b7837655f Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 26 Sep 2019 12:00:19 +0530
Subject: [PATCH 0292/1921] octeontx2-af: Fix default profile loading checks

Keysize of default profile is vaildated using a
variable which gets initialized later. Hence get
key size from register for the validation.

Change-Id: Ib44ebfc1c9c9c322ebc0e0228fe0b613333214d6
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/16520
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index 47874c7bb87e..ba0a24798ed9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -1003,7 +1003,6 @@ static void npc_config_rx_ldata_extract(struct rvu *rvu, int blkaddr)
 
 static void npc_config_ldata_extract(struct rvu *rvu, int blkaddr)
 {
-	struct npc_mcam *mcam = &rvu->hw->mcam;
 	int lid, ltype;
 	int lid_count;
 	u64 cfg;
@@ -1028,7 +1027,9 @@ static void npc_config_ldata_extract(struct rvu *rvu, int blkaddr)
 		}
 	}
 
-	if (mcam->keysize != NPC_MCAM_KEY_X2)
+	cfg = (rvu_read64(rvu, blkaddr, NPC_AF_INTFX_KEX_CFG(0)) >> 32) & 0x07;
+	/* Default profile works with key size NPC_MCAM_KEY_X2 */
+	if (cfg != NPC_MCAM_KEY_X2)
 		return;
 
 	/* Config RX ldata extract */
-- 
2.31.1

